lib/unittest.bzl

```diff
@@ -160,17 +163,39 @@ def _end(env):
     Args:
       env: The test environment returned by `unittest.begin`.
     """
-    cmd = "\n".join([
-        "cat << EOF",
-        "\n".join(env.failures),
-        "EOF",
-        "exit %d" % len(env.failures),
-    ])
+
+    if env.ctx.toolchains[_TOOLCHAIN_TYPE].bazel_skylib_toolchain_info.is_exec_windows:
```

Comment:
In general, having rules switch on a boolean in the toolchain is a bad code smell. Specifically, this ties your toolchain to "is windows?" and "is not windows?" checks.

Instead, would it be possible to have your toolchain declare several fields, one being a file extension (".bat" vs ".sh"), and another being templates for the success and failure messages, so that this function is reduced to just a single block with no if/else check?

Reply:
Done. Great suggestion, thank you!